### PR TITLE
[WIP] chore(node): Use node 12 for the auth/oauth servers.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
@linuxwolf suggests we migrate from node-jose to
https://github.com/panva/jose for our JW* needs
since it is more fully supported.

Moving to this library would also allow us to sign
JWTs using Ed25519 as well as make it simpler to
migrate to ES256. However, the lirary requires
node 12.

This is an attempt at using node 12 on the
OAuth and Auth servers.

issue #1797
issue #1794

### CODE REVIEW

**Comment by jbuck** [comment]
I should really switch these off alpine...

File: packages/fxa-auth-server/test/local/redis.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ describe('lib/redis:', () => {
     };
     fxaShared = sinon.spy(() => redis);
     wrapper = proxyquire(`${LIB_DIR}/redis`, {
-      '../../../fxa-shared/redis': fxaShared,
+      '../../fxa-shared/redis': fxaShared,
```
Comment (shane-tomlinson) [inline packages/fxa-auth-server/test/local/redis.js]: I'm not really sure why this change was necessary, I haven't looked to see if node 12 resolves modules differently to node 10.

File: packages/fxa-customs-server/package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     "format": "prettier '**' --write"
   },
   "dependencies": {
-    "@google-cloud/pubsub": "0.24.1",
+    "@google-cloud/pubsub": "^0.31.0",
```
Comment (shane-tomlinson) [inline packages/fxa-customs-server/package.json]: The previous version of pubsub was incompatible with node 12 and this update fixed that.

**Comment by shane-tomlinson** [comment]
Closing this until I have more time.
